### PR TITLE
Added new github workflow to use github actions for CI

### PR DESCRIPTION
**Pull request by mraarif**
**JIRA:**
https://openedx.atlassian.net/browse/BOM-2121

### CODE REVIEW

**Comment by mraarif** [comment]
@jmbowman should we start adding Django 3.1 tox environments?

**Comment by jmbowman** [comment]
In tox, sure.  For CI we should wait until we have time allocated to get them passing.

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py35-django22-drf{39,latest},py38-django{22,30}-drf{39,latest},quality, docs
+envlist = py35-django22-drf{39,latest},py38-django{22}-drf{39,latest},py38-django{30,31}-drf{39,latest},quality, docs
```
Comment (jmbowman) [inline tox.ini:2]: Was there a reason the Django versions needed to be split apart in separate entries like this?

Comment (mraarif) [reply tox.ini:2]: it should be `py38-django{22}-drf{39,latest},py38-django{30,31}-drf{latest}` I updated the `ci.yml` but forgot this, `djangorestframework` version 3.9 supports Django only up to 2.2 so we we've to exclude DRF 3.9 environment for Django 3.0 and 3.1

File: .github/workflows/pypi-publish.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install pip
+        run: pip install -r requirements/pip.txt
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPi
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: edx
```
[the remark this answers is not in the document]
Comment (mraarif) [reply .github/workflows/pypi-publish.yml:29]: I've added a question in [#travis-to-github-actions-migration](https://edx-internal.slack.com/archives/C01ES9LPGNR) regarding this

File: requirements/base.txt [hunk 3]
```diff
@@ -4,32 +4,33 @@
...
+certifi==2020.11.8        # via requests
+cffi==1.14.4              # via cryptography
 chardet==3.0.4            # via requests
-cryptography==3.1.1       # via pyjwt
+cryptography==3.2.1       # via pyjwt
+django-crum==0.7.9        # via edx-django-utils
 django-waffle==2.0.0      # via -r requirements/base.in, edx-django-utils
-django==2.2.16            # via -c requirements/constraints.txt, -r requirements/base.in, djangorestframework, drf-jwt, edx-django-utils, rest-condition
-djangorestframework==3.11.1  # via -r requirements/base.in, drf-jwt, rest-condition
+django==2.2.17            # via -c requirements/constraints.txt, -r requirements/base.in, django-crum, djangorestframework, drf-jwt, edx-django-utils, rest-condition
```
[the remark this answers is not in the document]
Comment (mraarif) [reply requirements/base.txt:13]: I think the reason for this is no one was tagged on the PR, I saw Tim has done some cleanup in upgrade and other jobs recently, we can look into the reason why appropriate teams are not being tagged.

File: Makefile [hunk 4]
```diff
@@ -34,6 +35,7 @@ upgrade: upgrade-piptools piptools ## upgrade requirement pins.
 	rm requirements/test.txt.tmp
 
 requirements: piptools ## install dev requirements into current env
+	pip install -qr requirements/pip.txt
 	pip-sync requirements/dev.txt
```
[the remark this answers is not in the document]
Comment (mraarif) [reply Makefile:40]: I think it calls `pip-tools` target first and that target is installing `pip-tools.txt`

[the remark this answers is not in the document]
Comment (jmbowman) [reply Makefile:39]: Ah, but then shouldn't the `pip.txt` installation go in the `pip-tools` target so as to be installed before trying to install pip-tools, etc.?

File: requirements/base.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmbowman) [reply requirements/base.txt:13]: arbi-bom didn't have explicit write permission to the repo, so team review requests were failing.  Just fixed that.

File: .github/workflows/pypi-publish.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.5
```
Comment (jmbowman) [inline .github/workflows/pypi-publish.yml:18]: Can we change this to 3.8?

File: Makefile [hunk 6]
```diff
@@ -22,6 +22,7 @@ upgrade-piptools: piptools # upgrade pip-tools using pip-tools.
 
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: upgrade-piptools piptools ## upgrade requirement pins.
+	pip-compile --allow-unsafe --rebuild -o requirements/pip.txt requirements/pip.in
 	pip-compile requirements/base.in --rebuild --upgrade -o requirements/base.txt
 	pip-compile requirements/test.in --rebuild --upgrade -o requirements/test.txt
 	pip-compile requirements/docs.in --rebuild --upgrade -o requirements/docs.txt
```
Comment (jmbowman) [inline Makefile:28]: We should only need `--rebuild` on the first pip-compile run in the upgrade target.

File: .github/workflows/pypi-publish.yml (hunk 5 above)
Comment (mraarif) [reply .github/workflows/pypi-publish.yml:18]: it is being used by `edx-proctoring` that is published to PyPi using python 3.5

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (mraarif) [reply Makefile]: moved `pip.txt` installation to `pip-tools` target.

File: .github/workflows/pypi-publish.yml (hunk 5 above)
Comment (jmbowman) [reply .github/workflows/pypi-publish.yml:18]: I'm not aware of any problems with using a package created and uploaded with a newer python version in an older one.  I'd like to minimize our usage of an unsupported python version.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**